### DISCUSSION TITLE
Italian dictionary

### DISCUSSION THREAD

**Discussion by robopernext** [tiredofit/docker-collabora-online#15] [General]
Hi,
congratulations on this docker, it is very useful.
I am Italian and it would be useful for me to have the dictionary in Italian.
Can you add the Italian dictionary in the next release?
Thanks
Roberto

**Comment by tiredofit** [comment]
I'll either add it or show you how to do it. Have you figured out how to do it with the current image?

**Reply by robopernext** [reply]
No I don't know how to do that. I am not very skilled in docker. If you can show me how to do it, I will try to insert the Italian dictionary in the current immage.
Thanks.

**Reply by tiredofit** [reply]
ok! lets experiment. 

Enter inside the image `docker exec -it (your container name) bash`
Reload the package repositories: `apt-get update`
Install the Italian spell checker `apt-get install -y hunspell-it`
Reload LibreOffice Online (dirty): `pkill loolwsd`

Theoretically that should give you Italian support, but I haven't tested.
If this works, then I can help you build a script to do this every time the container starts up rather than me building it into the image, and also avoiding you having to build the image as it takes quite a long time and uses quite alot of resources.

**Reply by robopernext** [reply]
Hi, these are the steps I followed:
- docker exec -t -i container-ID /bin/bash
- apt update && apt install hunspell-it
- cp /usr/share/hunspell/it* /opt/libreoffice/share/extensions/dict-it/
- exit from shell
- docker commit container-ID image-name

In this way I have a new image with the Italian dictionary.

To make your docker more international, my advice is to include all dictionaries in future versions.

Thanks for your support, I will continue to follow your libreoffice-online docker updates.

**Reply by tiredofit** [reply]
Thanks for adding the final piece in there to get the dictionary to load. I've taken those steps and created a new release `2.2.0` (Building, should be on Docker Hub in 12 hours) that downloads and installs your dictionary for you. Have a peek at the README and see the section regarding spellcheck.

I've opted to not install all dictionaries on container build due to the size that it would bring the image to, this option at least saves you the step of hacking away at the image each time I have a release.

**Reply by robopernext** [reply]
I tried to start with environment variable DICTIONARIES="it,it_IT,en_GB,en_US" but the variable is not considered.

This is the log:
2021-07-12-05:40:50 [NOTICE] ** [libreoffice-online] Setting up custom dictionary support - Please wait..
2021-07-12-05:40:52 [NOTICE] ** [libreoffice-online] Adding dictionary for '{lang}' - Please wait..

So the dictionaries are not installed.

**Reply by tiredofit** [reply]
Oops! Definitely a problem on that notice, but the dictionaries are being installed - You can see the it happening if you add CONTAINER_LOG_LEVEL=DEBUG as an environment variable.
